openedx/openedx-platform
# [WIP] ENT-304 Add confirm email field to registration

As part of our efforts to improve the registration page, we are adding an email confirmation field.

**Sandbox:** https://brittneyexline.sandbox.edx.org/
Note that this sandbox already has the below configuration set. The staff user has admin access.

**Testing instructions**
1. With no configuration set, visit the registration page. It should behave the same as it does.
2. Go into the Site Configuration in django admin and add a configuration with the following value: 
`{
  "ENABLE_COMBINED_LOGIN_REGISTRATION":true,
  "REGISTRATION_EXTRA_FIELDS":{
    "confirm_email":"required",
    "city":"hidden",
    "country":"hidden",
    "gender":"optional",
    "goals":"optional",
    "honor_code":"required",
    "level_of_education":"optional",
    "mailing_address":"optional",
    "year_of_birth":"optional"
  },
  "REGISTRATION_FIELD_ORDER":[
    "email",
    "confirm_email",
    "name",
    "username",
    "password",
    "first_name",
    "last_name",
    "city",
    "state",
    "country",
    "gender",
    "year_of_birth",
    "level_of_education",
    "company",
    "title",
    "mailing_address",
    "goals",
    "honor_code",
    "terms_of_service"
  ]
}`
3. Restart edxapp.
4. Visit the registration page. The order should now reflect the order specified. There should also be a confirm email field. If the form is filled out without the confirm email field matching the email field, there should be an error message asking to confirm your email. If they match, registration should be successful.
5. Adjust the configuration field order and observe that the registration form adjusts field order accordingly. If a field is removed from field order, or if an unknown field is added, the order will revert to the existing order.
